src/main/java/org/prebid/server/bidder/pubmatic/PubmaticBidder.java

```diff
@@ -1,30 +1,259 @@
+        }
+
+        modifySiteOrApp(pubId, bidRequest, requestBuilder);
+
+        final BidRequest outgoingRequest = requestBuilder.build();
+        final String body = Json.encode(outgoingRequest);
+
+        return Result.of(
+                Collections.singletonList(HttpRequest.of(HttpMethod.POST, endpointUrl, body, BidderUtil.headers(),
+                        outgoingRequest)), errors);
+    }
+
+    private void modifySiteOrApp(String pubId,
+                                 BidRequest bidRequest,
+                                 BidRequest.BidRequestBuilder bidRequestBuilder) {
+        if (bidRequest.getSite() != null) {
+            final Site site = bidRequest.getSite();
+            if (site.getPublisher() != null) {
+                final Publisher modifiedPublisher = site.getPublisher().toBuilder().id(pubId).build();
+                bidRequestBuilder.site(site.toBuilder().publisher(modifiedPublisher).build());
+            } else {
+                bidRequestBuilder.site(site.toBuilder()
+                        .publisher(Publisher.builder().id(pubId).build())
+                        .build());
+            }
+        } else if (bidRequest.getApp() != null) {
+            final App app = bidRequest.getApp();
+            if (app.getPublisher() != null) {
+                final Publisher modifiedPublisher = app.getPublisher().toBuilder().id(pubId).build();
+                bidRequestBuilder.app(app.toBuilder().publisher(modifiedPublisher).build());
+            } else {
+                bidRequestBuilder.app(app.toBuilder()
+                        .publisher(Publisher.builder().id(pubId).build())
+                        .build());
+            }
+        }
+    }
+
+    private String getWrapExt(Imp imp, ExtImpPubmatic extImpPubmatic) {
+        try {
+            Json.mapper.convertValue(extImpPubmatic.getWrapExt(), new TypeReference<Map<String, Integer>>() {
+            });
+        } catch (IllegalArgumentException e) {
+            throw new PreBidException(String.format("Error in Wrapper Parameters = %s  for ImpID = %s WrapperExt = %s",
+                    e.getMessage(), imp.getId(), extImpPubmatic.getWrapExt().toString()));
+        }
+        return extImpPubmatic.getWrapExt().toString();
+    }
+
+    private Imp parseImpObject(Imp imp, ExtImpPubmatic extImpPubmatic) {
+        final Imp.ImpBuilder result = imp.toBuilder();
+
+        if (imp.getBanner() == null && imp.getVideo() == null) {
+            throw new PreBidException(String.format("Invalid MediaType. PubMatic only supports Banner and Video. "
+                    + "Ignoring ImpID=%s", imp.getId()));
+        }
+        if (imp.getAudio() != null) {
+            result.audio(null);
+        }
+
+        final String adSlotString = extImpPubmatic.getAdSlot().trim();
+        final String[] adSlot = adSlotString.split("@");
+
+        if (imp.getBanner() != null) {
+            if (adSlot.length == 2 && !adSlot[0].equals("") && !adSlot[1].equals("")) {
```

Comment:
1. Use `StringUtils.isBlank(()` instead of `equals(...)`
2. Use fail-first principal for conditions:
```
            if (adSlot.length != 2 || StringUtils.isBlank(adSlot[0]) || StringUtils.isBlank(adSlot[1])) {
                throw new PreBidException("Invalid adSlot provided");
            }
```